app/lib/search/text_utils.dart

```diff
@@ -37,3 +37,15 @@ Iterable<String> splitForIndexing(String text) {
 
 List<String> extractExactPhrases(String text) =>
     _exactTermRegExp.allMatches(text).map((m) => m.group(1)).toList();
+
+class StringInternPool {
```

Comment:
There is already a class for this. Also as mentioned before, please use "canonicalize" instead of "interning" (which nobody knows what it means).

app/lib/shared/deps_graph.dart:15:class StringCanonicalization {

Reply:
Ah, good point about that class, it shall be a single one of it.

However, I disagree with the name, as both Java and C# are using it for a long time:
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#intern()
https://msdn.microsoft.com/en-us/library/system.string.intern.aspx

Canonicalization changes the date (like in a path `/a/../b/` -> `/b/`), while interning does not change it. There are more examples here: https://en.wikipedia.org/wiki/Canonicalization